support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -28,14 +32,20 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     context: Context,
     services: Services
   ): FutureHandlerResult = {
+    val now = () => OffsetDateTime.now
     val subscribeItem = buildSubscribeItem(state, services.promotionService)
     for {
       identityId <- Future.fromTry(IdentityId(state.user.id))
-      maybeDomainSubscription <- GetSubscriptionWithCurrentRequestId(services.zuoraService, state.requestId, identityId, state.product.billingPeriod)
+        .withLogging("identity id")
+      maybeDomainSubscription <- GetSubscriptionWithCurrentRequestId(services.zuoraService, state.requestId, identityId, state.product.billingPeriod, now)
+          .withLogging("GetSubscriptionWithCurrentRequestId")
       previewPaymentSchedule <- PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, services, checkSingleResponse)
+          .withLogging("PreviewPaymentSchedule")
       thankYouState <- maybeDomainSubscription match {
         case Some(domainSubscription) => skipSubscribe(state, requestInfo, previewPaymentSchedule, domainSubscription)
+            .withLogging("skipSubscribe")
         case None => subscribe(state, subscribeItem, services).map(response => toHandlerResult(state, response, previewPaymentSchedule, requestInfo))
+          .withLogging("subscribe")
```

Comment:
this seems like a lot of extra logging, will we see it in PROD?

Reply:
yes we will, I'm open to adjusting the logging, but my usual problem is not enough logging when looking at support workers, hence I was generous

Reply:
doesn't look bad so far https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logEventViewer:group=/aws/lambda/support-CreateZuoraSubscriptionLambda-PROD;stream=2019/08/14/[$LATEST]d92a77593ac54659a5acca38c8b8c74d;start=2019-08-13T13:40:58Z